cilium/CFP-39757-doc-improvement-plan.md

```diff
@@ -0,0 +1,620 @@
+types.
+
+Each page title is based on some kind of concept. For any substantial topic,
+include a general concepts page named "General concepts" whose file name is
+**general.rst**. This is the page to link to whenever we want to refer to the
+topic in a general way.
+
+Prefix the topic to the index page's title. For example:
+
+* "TLS inspection concepts"
+
+The section should render as the following in the menu:
+
+* "Concepts" (same as a standalone concepts page)
+
+## Reference section
+
+Use a reference section in the navigation menu, and thus a **reference**
+directory, and an **index.rst** file, if you have multiple reference page
+types.
+
+Each page title is based on some kind of reference.
+
+Prefix the topic to the index page's title. For example:
+
+* "TLS inspection reference"
+
+The section should render as the following in the menu:
+
+* "Reference" (same as a standalone reference page)
+
+## Resources section
+
+Any single resource page should be accompanied by a resources section in the
+navigation menu, and thus a **resources** directory, and an **index.rst** file.
+This is in contrast to conceptual or reference page types where a section is
+only needed for multiple such pages.
+
+When you have content under a "Resources" sub-section (in a How-to) that grows
+too large (about 1/2 a page), place it in this section as separate files.
+
+When there are multiple How-to pages under a given topic, to avoid repeating
+the same "Resources" material in each, you can optionally place it in this
+section regardless of its size.  You can also use this section when there is
+docsuggest content that applies to a topic in general.
+
+Prefix the topic to the index page's title. For example:
+
+* "TLS inspection resources"
+
+The section should render as the following in the menu:
+
+* "Resources"
+
+Each of its pages' names are based on a docsuggest. For example:
+
+* "Caveats and limitations"
+* "Troubleshooting"
+
+The file names are:
+
+* **caveats-and-limitations.rst**
+* **troubleshooting.rst**
+
+The rendered navigation menu looks like:
+
+    	`+ Resources`<br>
+    	`+ Caveats and limitations`<br>
+    	`+ Troubleshooting`
+
+## Media
+
+Create a directory under the topic directory called **media** to contain media
+files.
+
+## Ordering of items
+
+Maintain the relative ordering of items whether they are sub-sections within a
+how-to or pages within a navigation section.
+
+### Sub-sections within a how-to
+
+After the main body of a how-to there is a section for resources:
+
+* Resources
+* Caveats and limitations
+* Monitoring
+* Security
+* Upgrade and downgrade
+* Failure scenarios
+* Scalability testing
+* Troubleshooting
+* Disable
+* Removal
+* Further reading
+
+### Pages in the navigation menu
+
+For a given topic, this is the page order as they appear in the menu:
+
+**How-to** pages
+
+**Conceptual** pages
+
+**Reference** pages
+
+**Resources** pages
+
+## Appendix A - Software feature readiness levels
```

Comment:
FWIW I think this is fine as a supporting appendix of this CFP, but more generally for @cilium/committers we may want to draft a separate CFP to formalize these feature readiness levels more generally beyond the context of the docs.